.github/workflows/womtools-validate.yml

```diff
@@ -1,15 +1,36 @@
 name: WOMtool Validation of WDL Script
 
 on:
+  pull_request:
   workflow_dispatch:
 
 permissions:
   contents: read
   packages: write
 
 jobs:
+  generate-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Find WDL Directories
+        id: set-matrix
+        run: |
+          # Find all directories containing .wdl files
+          dirs=$(find . -type f -name "*.wdl" -exec dirname {} \; | sort -u | sed 's|^\./||' | jq -R . | jq -s | jq -c .)
+          echo "Directories: $dirs"
+          echo "matrix=$dirs" >> $GITHUB_OUTPUT
+
   womtoolval:
+    if: ${{ github.event.pull_request.head.repo.fork == false }}
```

Comment:
I think you need to add the `needs: generate-matrix` here.

Reply:
absolutely, thank you